janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -748,12 +748,12 @@ private SearchParams convertQuery(Condition<?> condition, final KeyInformation.S
                     } else if (janusgraphPredicate == Cmp.EQUAL || janusgraphPredicate == Cmp.NOT_EQUAL) {
                         tokenize(params, map, delegatingAnalyzer, (String) value, stringFieldKey, janusgraphPredicate);
                     } else if (janusgraphPredicate == Text.FUZZY) {
-                        params.addQuery(new FuzzyQuery(new Term(stringFieldKey, (String) value)));
+                        params.addQuery(new FuzzyQuery(new Term(stringFieldKey, (String) value), Text.getMaxEditDistance((String) value)));
```

Comment:
I am just thinking if we should make JanusGraph's Text.CONTAINS_FUZZY respect Lucene's default behaviour or vice versa. Any idea?

Either way, it would be a breaking change, so we need to explain it in the changelog.md.

Reply:
Either way works, I just wanted them to be consistent in both places. I only use JanusGraph with Elastic, which already were aligned.

I updated this so the generic testing I added worked across the board, but Ill defer to you for deciding if the change should be JG/ES or Lucene/Solr.

Ive found the default edit distance of 2 to cause some bad matching on shorter terms, which is another reason I reused JG/ES methodology.

Reply:
Sounds good to me. We just need to mention this in the changelog since this is a breaking change.

Reply:
Perfect, updated the changelog.md in the last merge. Let me know if that looks good to you, I just added a note to the end of the existing list for 0.6.0.

Reply:
> It might make sense to be explicit in the ES code instead of using AUTO, which might change implementation at some point in the future.

Is it possible to continue using AUTO, but have some test cases (if we don't have yet) to detect potential implementation change in the future? That way, we would be able to enforce the current behavior and can make a call if a future ES version changes this behavior.

Reply:
Yes! The generic predicate tests Ive added to IndexProviderTest already covers this, as the @MethodSource arguments provided by TextArgument include these cases.